app/extensions/logging/__init__.py

```diff
@@ -4,6 +4,45 @@
 ---------------
 """
 import logging
+import enum
+from flask_login import current_user  # NOQA
+
+
+class AuditType(str, enum.Enum):
+    Create = 'Create'
+    Delete = 'Delete'
+    Update = 'Update'  # Generic Update
+
+
+# somewhere between Error and Critical
+AUDIT = 45
+
+
+# logger for calling file needed as a parameter to ensure that the file and line numbers are correct in logs
+def audit_log(logger, msg, *args, **kwargs):
+    assert object
+
+    # First iteration. Timestamp added by logger so no need to add manually
+    if current_user is not None and not current_user.is_anonymous:
+        msg = f'{msg} executed by user :{current_user.guid} {current_user.email}'
+    else:
+        msg = f' {msg} executed by anonymous user'
+    logger.log(AUDIT, msg, *args, **kwargs)
+
+
+# As per above but this time an object must be passed that must have a guid member
+def audit_log_object(logger, obj, audit_type, msg, *args, **kwargs):
+    assert obj
+    assert hasattr(obj, 'guid')
+    assert isinstance(audit_type, AuditType)
+
+    msg = f'{audit_type} of {obj.__class__.__name__} {obj.guid} {msg}'
```

Comment:
i think this is good enough for a start, but i wonder how "filterable" this will be for later?  e.g. if i am wanting to find logs on object class X, guid Y ... will this involve grepping/matching against the log-message content?   maybe this is fine and the best way forward (given the constraints of `logger`).

Reply:
For now it will yes. The idea is that drop 1 is entries in the log. Drop 2 is each of these become entries in a DB that we can then search.